FactoryPattern_API/Activity.cs

```diff
@@ -6,26 +6,51 @@
 
 namespace FactoryPattern_API
 {
-    class Activity : IProduct
+    class Activity : Products
     {
         FileDatabase file = new FileDatabase();
```

Comment:
incorrect way of intializing where you want to save , this creates a dependency on filesystem for this class. Ideally it should be in form of an interface